doc/prelims.rst

```diff
@@ -22,39 +25,75 @@ Running a **Linux** distribution, Python must be compiled. Download and compile
+
+.. note::
+
+   Remember to use the correct call to python/python3/python3.6 and pip/pip3/pip3.6 depending how your Python 3.6 was installed. This is neccesary to install software in the correct locations for the remainder of these instructions.
+
+
+2) Install Clang and virtualenvwrapper
+--------------------------------------
+
+a) Install/Update Clang (macOS Only)
+************************************
+
+`Clang <https://clang.llvm.org/>`__ is a compiler front-end that supports multiple languages. Although an older version is already installed on your system, install the newest version using brew:
+
+.. code-block:: shell
+
+   brew install llvm
+
+To make this most recent version of clang the default compiler on your system (and enable OpenMP support) paste the following lines into the .profile file created above:
+
+.. code-block:: shell
+
+   export CC=/usr/local/Cellar/llvm/4.0.0_1/bin/clang-4.0
+   export CXX=/usr/local/Cellar/llvm/4.0.0_1/bin/clang++
+   export LDFLAGS=-L$(brew --prefix llvm)/lib
+
+OpenMP is packaged with the new versions of the clang compiler. This tool allows for shared memory multiprocessing in C and C++; in the context of this software OpenMP allows for parallel track generation during the minimization and is highly reccomended for running the analyses.
+
+b) Install virtualenvwrapper
+****************************
 
 Install and setup virtualenvwrapper and associated tools. This allows you to create isolated "virtual environments" with independent installations of Python packages. This isn't strictly necessary, but helps prevent conflicts between incompatible package versions. To install virtualenvwrapper, run the command below:
 
 .. code-block:: shell
    
-   pip3.6 install virtualenvwrapper
+   pip install virtualenvwrapper   # sudo might be required
 
-An introduction and walkthrough to using the virtualenvwrapper tool is found `here <https://virtualenvwrapper.readthedocs.io/en/latest/>`__. To use virtual environments, place the following lines in the .bashrc file. The first line sets the Python interpreter for your virtual environments to python3.6. The last line is a path to your shell startup file, change it depending on where virtualenvwrapper was installed by pip.
+An introduction and walkthrough to using the virtualenvwrapper tool can be found `here <https://virtualenvwrapper.readthedocs.io/en/latest/>`__. To use virtual environments, place the following lines in your .bashrc or .profile file. The first line sets the Python interpreter for your virtual environments to python3.6. The last line is a path to your shell startup file and you should change it depending on where virtualenvwrapper was installed by pip.
 
 .. code-block:: shell
 
    VIRTUALENVWRAPPER_PYTHON=/usr/local/bin/python3.6
```

Comment:
This path is not correct on the Mac if you install Python from the dmg file from python.org. That installer puts it somewhere deep in `/Library/Frameworks`. Maybe we should write `VIRTUALENVWRAPPER_PYTHON=$(which python3.6)` instead.